compiler/rustc_mir/src/interpret/eval_context.rs

```diff
@@ -783,20 +778,20 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             throw_ub_format!("unwinding past the topmost frame of the stack");
         }
 
-        let frame =
-            self.stack_mut().pop().expect("tried to pop a stack frame, but there were none");
-
         if !unwinding {
             // Copy the return value to the caller's stack frame.
-            if let Some(ref return_place) = frame.return_place {
-                let op = self.access_local(&frame, mir::RETURN_PLACE, None)?;
-                self.copy_op_transmute(&op, return_place)?;
-                trace!("{:?}", self.dump_place(**return_place));
+            if let Some(return_place) = self.frame().return_place {
+                let op = self.access_local(self.frame(), mir::RETURN_PLACE, None)?;
+                self.copy_op_transmute(&op, &return_place)?;
+                trace!("{:?}", self.dump_place(*return_place));
             } else {
                 throw_ub!(Unreachable);
             }
         }
 
+        let frame =
+            self.stack_mut().pop().expect("tried to pop a stack frame, but there were none");
```

Comment:
The order of operations here is very tricky, so if possible this would be easier to review if you wouldn't have to reorder this.^^